librz/include/rz_asm.h

```diff
@@ -91,6 +91,17 @@ typedef struct {
 	char *value;
 } RzAsmEqu;
 
+/**
+ * \brief Holds several settings for plugins.
+ */
+typedef struct {
+	bool immdisp; //< Display immediates with # symbol (for arm/hexagon architectures). false = show hashs
+	bool immsign; //< Print signed immediates as negative values, not their unsigned representation.
+	bool utf8; //< Flag for plugins: Use utf-8 characters.
+	bool hex_sdk; //< Hexagon arch only. Print packet syntax in hexagon-objdump style.
+	bool hex_reg_alias; //< Print the alias of registers (Alias from C0 = SA0).
+} RzAsmPluginSettings;
```

Comment:
How about using a RzConfig instead of this?
This struct could be passed as argument when initializing the asm plugin, so you can add `plugins.hexagon.sdk` and `plugins.hexagon.reg.alias` dynamically (kinda like i do for jsdec).
I'm proposing this because i hate to have duplicates like we are having here.

@thestr4ng3r @ret2libc @XVilka

Reply:
Good idea about this one.

Reply:
The plugins should not be given a `RzConfig` to actively mess with imo. What I would imagine instead is the plugin declaring in some data structure that can be queried that it wants a config var e.g. `reg.alias` bound to some callback and then core/asm will automatically sort that in as `plugins.hexagon.reg.alias` into its `RzConfig`.
The idea is also that you will use these plugins without a `core` in the future, but maybe such a refactoring is best done in combination with `RzArch`.

Reply:
This sounds really cool.

Maybe something like:

Add callbacks in plugin:
```c
RZ_BORROW RzConfig *get_plugin_config(); // Returns plugins RzConfig object with config nodes it would like to set.
void update_config_node(RzConfigNode *node); // Updates value in the plugins RzConfig
```

Before the plugin is put in use the core does:
- `config = RzAsmPlugin::get_plugin_config()`
- Iterates of each `node` in `config`
  - Does `SETCB("plugin.<plugin_name>" + node.conf, <default>, update_config_node, node.desc)`.

And if the plugin is disabled:
- `config = RzAsmPlugin::get_plugin_config()`
- Iterates of each `node` in `config`
  - Do `UNSETCB("plugin.<plugin_name>" + node.conf)`.

Reply:
This would also have the advantage that the plugin specific settings don't clutter the enormous amount of settings if not in use.

Reply:
this could be even better. i like the proposal from @Rot127

Reply:
@thestr4ng3r Any comments on the proposal? Otherwise would implement it tomorrow.

Reply:
Seems like an interesting idea, if it works. Currently asm does not depend on config though, but since config only uses util, might be ok to add.